services/QuillLMS/app/models/student_response.rb

```diff
@@ -0,0 +1,257 @@
+#  index_student_responses_on_activity_session_id                  (activity_session_id)
+#  index_student_responses_on_question_id                          (question_id)
+#  index_student_responses_on_student_response_answer_text_id      (student_response_answer_text_id)
+#  index_student_responses_on_student_response_directions_text_id  (student_response_directions_text_id)
+#  index_student_responses_on_student_response_prompt_text_id      (student_response_prompt_text_id)
+#  index_student_responses_on_student_response_question_type_id    (student_response_question_type_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (activity_session_id => activity_sessions.id)
+#  fk_rails_...  (student_response_answer_text_id => student_response_answer_texts.id)
+#  fk_rails_...  (student_response_directions_text_id => student_response_directions_texts.id)
+#  fk_rails_...  (student_response_instructions_text_id => student_response_instructions_texts.id)
+#  fk_rails_...  (student_response_previous_feedback_text_id => student_response_previous_feedback_texts.id)
+#  fk_rails_...  (student_response_prompt_text_id => student_response_prompt_texts.id)
+#  fk_rails_...  (student_response_question_type_id => student_response_question_types.id)
+#
+class StudentResponse < ApplicationRecord
+  belongs_to :activity_session
+  belongs_to :question
+  belongs_to :student_response_answer_text
+  belongs_to :student_response_directions_text
+  belongs_to :student_response_instructions_text
+  belongs_to :student_response_previous_feedback_text
+  belongs_to :student_response_prompt_text
+  belongs_to :student_response_question_type
+
+  has_one :student_response_extra_metadata, dependent: :destroy
+
+  has_many :student_response_concept_results, dependent: :destroy
+  has_many :concept_results, through: :student_response_concept_results
+
+  has_many :student_responses_concepts, dependent: :destroy
+  has_many :concepts, through: :student_responses_concepts
+
+
+  validates_exclusion_of :correct, in: [nil]
+
+  # This is a list of keys from the old ConceptResults records
+  # that this model knows how to normalize.  Any keys that are
+  # fed into create_from_json that aren't from this list,
+  # will be stashed in a related StudentResponseExtraMetadata
+  # record
+  KNOWN_METADATA_KEYS = [
+    :answer,
+    :attemptNumber,
+    :correct,
+    :directions,
+    :lastFeedback,
+    :prompt,
+    :questionNumber,
+    :questionScore
+  ]
+
+  def self.create_from_json(data_hash)
+    data_hash = data_hash.deep_symbolize_keys
+
+    metadata = data_hash[:metadata]
+
+    question = calculate_question_from_hash(data_hash)
+
+    # Make sure that singular concept_id gets converted to array of
+    # conept_ids if that hasn't been done by an earlier step
+    data_hash[:concept_ids] = [data_hash[:concept_id]] unless data_hash.key?(:concept_ids) && !data_hash[:concept_ids].empty?
+
+    extra_metadata = extract_extra_metadata(metadata)
```

Comment:
There's a mix of class methods/variables, that likely should be instance variables/methods, e.g. you might consider setting an `attr_accessor` of `metadata` on the student_response model, then you could set everyone on the instance from that.